### PR TITLE
fix: arrow element TS definition

### PR DESCRIPTION
**Pull request by gynekolog**
Arrow element has type HTMLElement | undefined. That cause a TS error with const `arrowRef = useRef(null)`.
If you use useRef without init value (`arrowRef = useRef()`) you got TS error on the target element.

image

image

**Solution**
Change arrow element type to `HTMLElement | null`.

reproduction: https://codesandbox.io/s/floating-ui-react-dom-visual-test-05xpu?file=/src/App.tsx

**Notes**
You can see in the example that there are other errors. It copy-pasted version of the https://github.com/floating-ui/floating-ui/blob/master/packages/react-dom/test/visual/index.js with annotated `arrowRef`. This pull request takes care only of the arrow element, not other errors.

### CODE REVIEW

File: packages/react-dom/index.test-d.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import {useFloating, shift, arrow} from '.';
 
 App;
 function App() {
-  const arrowRef = useRef();
+  const arrowRef = useRef(null);
```
Comment (atomiks) [inline packages/react-dom/index.test-d.ts:6]: should add the arrow JSX `ref={arrowRef}` to test that as well

Comment (gynekolog) [reply packages/react-dom/index.test-d.ts:6]: @atomiks I want but I'm not sure how can do it. Will you be fine with renaming `index.test-d.ts` to `index.test-d.tsx` and returning the arrow element?:
```tsx
function App() {
  const arrowRef = useRef(null);
  ...
  return (<div ref={arrowRef} />);
}
```

Comment (atomiks) [reply packages/react-dom/index.test-d.ts:6]: Yep that works! as long as `tsc` command typechecks `.tsx`

**Review by atomiks — COMMENTED** [review]
React Native needs the same changes

**Comment by atomiks** [comment]
Thanks:)

**Pull request merged**